src/Tools/dotnet-counters/Exporters/ConsoleWriter.cs

```diff
@@ -144,38 +157,64 @@ public void AssignRowsAndInitializeDisplay()
                 Console.WriteLine(_errorText);
                 row += GetLineWrappedLines(_errorText);
             }
-            Console.WriteLine(); row++; // Blank line.
 
-            foreach (ObservedProvider provider in _providers.Values.OrderBy(p => p.KnownProvider == null).ThenBy(p => p.Name)) // Known providers first.
+            bool RenderRow(ref int row, string lineOutput = null, Action rowAssignment = null)
             {
-                Console.WriteLine($"[{provider.Name}]"); row++;
+                if (row >= _consoleHeight + _topRow) // prevents from displaying more counters than vertical space available
+                {
+                    return false;
+                }
+
+                if (lineOutput != null)
+                {
+                    Console.Write(lineOutput);
+                }
+
+                if (row < _consoleHeight + _topRow - 1) // prevents screen from scrolling due to newline on last line of console
+                {
+                    Console.WriteLine();
+                }
+
+                rowAssignment?.Invoke();
+
+                row++;
 
-                foreach (ObservedCounter counter in provider.Counters.Values.OrderBy(c => c.DisplayName))
+                return true;
+            }
+
+            if (RenderRow(ref row)) // Blank line.
+            {
+                foreach (ObservedProvider provider in _providers.Values.OrderBy(p => p.KnownProvider == null).ThenBy(p => p.Name)) // Known providers first.
                 {
+                    if (!RenderRow(ref row, $"[{provider.Name}]"))
+                    {
+                        break;
+                    }
 
-                    string name = MakeFixedWidth($"{new string(' ', Indent)}{counter.DisplayName}", Indent + _maxNameLength);
-                    counter.Row = row++;
-                    if (counter.RenderValueInline)
+                    foreach (ObservedCounter counter in provider.Counters.Values.OrderBy(c => c.DisplayName))
                     {
-                        if (row >= _consoleHeight) // prevents from displaying more counters than vertical space available
+                        string name = MakeFixedWidth($"{new string(' ', Indent)}{counter.DisplayName}", Indent + _maxNameLength);
+                        if (counter.RenderValueInline)
                         {
-                            break;
+                            if (!RenderRow(ref row, $"{name} {FormatValue(counter.LastValue)}", () => { counter.Row = row; }))
```

Comment:
Not terribly pressing given that counters isn't perf sensitive, but this will need to allocate to capture. Can this be Action<ObservedCounter>, make the lambda static, and pass in the counter as a parameter?